relay-server/src/metrics_extraction/transactions/mod.rs

```diff
@@ -736,6 +740,131 @@ fn extract_captured_substring<'a>(string: &'a str, pattern: &'a Lazy<Regex>) ->
+fn span_op_to_category(op: &str) -> Option<&str> {
+    Some({
+        // General
+        if op.starts_with("mark") {
+            "mark"
+        } else if op.starts_with("function") {
+            "function"
+        }
+        //
+        // Browser
+        else if op.starts_with("pageload") {
+            "pageload"
+        } else if op.starts_with("navigation") {
+            "navigation"
+        } else if op.starts_with("resource") {
+            "resource"
+        } else if op.starts_with("browser") {
+            "browser"
+        } else if op.starts_with("measure") {
+            "measure"
+        } else if op.starts_with("ui") {
+            "ui"
+        } else if op.starts_with("http") {
+            "http"
+        } else if op.starts_with("serialize") {
+            "serialize"
+        }
+        //
+        // JS frameworks
+        else if op.starts_with("ui.react") {
+            "ui.react"
+        } else if op.starts_with("ui.vue") {
+            "ui.vue"
+        } else if op.starts_with("ui.svelte") {
+            "ui.svelte"
+        } else if op.starts_with("ui.angular") {
+            "ui.angular"
+        } else if op.starts_with("ui.ember") {
+            "ui.ember"
+        }
+        //
+        // Web server
+        // `http*` mapped in Browser
+        else if op.starts_with("websocket") {
+            "websocker"
+        } else if op.starts_with("rpc") {
+            "rpc"
+        } else if op.starts_with("grpc") {
+            "grpc"
+        } else if op.starts_with("graphql") {
+            "graphql"
+        } else if op.starts_with("subprocess") {
+            "subprocess"
+        } else if op.starts_with("middleware") {
+            "middleware"
+        } else if op.starts_with("view") {
+            "view"
+        } else if op.starts_with("template") {
+            "template"
+        } else if op.starts_with("event") {
+            "event"
+        } else if op.starts_with("function.nextjs") {
+            "function.nextjs"
+        } else if op.starts_with("function.remix") {
+            "function.remix"
+        } else if op.starts_with("function") {
+            "function"
+        } else if op.starts_with("serialize") {
+            "serialize"
+        } else if op.starts_with("console") {
+            "console"
+        } else if op.starts_with("file") {
+            "file"
+        } else if op.starts_with("app") {
+            "app"
+        }
+        //
+        // Database
+        else if op.starts_with("db") {
+            "db"
+        } else if op.starts_with("cache") {
+            "cache"
+        }
+        //
+        // Serverless
+        // `http*` marked in Browser
+        // `grpc*` marked in Web server
+        else if op.starts_with("function.gpc") {
+            "function.grpc"
+        } else if op.starts_with("function.aws") {
+            "function.aws"
+        } else if op.starts_with("function.azure") {
+            "function.azure"
+        }
```

Comment:
you also might want to do it a bit differently, make the inner `if`s, something like 
```rust
// e.g. first match function and only then you can check if the function match something else, otherwise return `function`
else if op.starts_with("function") {
       if op.starts_with("function.gpc") {
            "function.grpc"
        } else if op.starts_with("function.aws") {
            "function.aws"
        } else if op.starts_with("function.azure") {
            "function.azure"
        } else {
            "function"
       }
}

```

what do you think?

Reply:
Nice! It also discovered a bug (see [ae68c82](https://github.com/getsentry/relay/pull/2181/commits/ae68c82364b7dacaf6fd7fe37ebbf54660a698b2)).